src/java.base/share/classes/java/util/Comparator.java

```diff
@@ -368,7 +368,7 @@ public static <T extends Comparable<? super T>> Comparator<T> reverseOrder() {
      * @since 1.8
      */
     @SuppressWarnings("unchecked")
-    public static <T extends Comparable<@NonNull ? super @NonNull T>> Comparator<T> naturalOrder() {
+    public static <T extends Comparable<? super T>> Comparator<T> naturalOrder() {
```

Comment:
The upper bound of `Comparable` is [NonNull](https://github.com/eisop/jdk/blob/f877f6dc69cf12c0dc40b9650f21f467a6b6ab02/src/java.base/share/classes/java/lang/Comparable.java#L113C57-L113C57) (the redundant `@NonNull` should be removed over there).
The type parameter `T` here has a `@NonNull` upper bound.
`T` is used as the lower bound of the wildcard and already `@NonNull`, so the extra annotation does nothing.
The upper bound of the wildcard is `@Nullable Object` which gets intersected with the upper bound of `Comparable`.
If anything, I would keep that annotation, that is, change it to `@NonNull ? super T`, to make it explicit that the type argument is within the bound.
But in the end I agree that the `@NonNull`s here look redundant.